### PR TITLE
chore(ci): auto generation of configs for deb packages

### LINKED ISSUE #1692: Installation on Ubuntu 21.10 Impish not working
## Description

Installing Trivy according to [official instruction here](https://aquasecurity.github.io/trivy/v0.21.2/getting-started/installation/#debianubuntu)

Installation via `apt` on the latest version of Ubuntu is not supported.

## What did you expect to happen?
Successfull installation

## What happened instead?
```
$ sudo apt-get update
Err:12 https://aquasecurity.github.io/trivy-repo/deb impish Release
  404  Not Found [IP: 185.199.110.153 443]
```

### PR DESCRIPTION
**Pull request by afdesk**
## Description
Added auto generation `conf/distributions` for building deb packages.

## Related issues
- Close #1692 

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy/blob/main/CONTRIBUTING.md) to this repository.
- [x] I've followed the [conventions](https://github.com/aquasecurity/trivy/blob/main/CONTRIBUTING.md#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: ci/deploy-deb.sh [hunk 1]
```diff
@@ -12,6 +12,20 @@ for release in ${DEBIAN_RELEASES[@]} ${UBUNTU_RELEASES[@]}; do
   reprepro -A arm64 remove $release trivy
 done
 
+rm -r conf
+mkdir conf
+
+for release in ${UBUNTU_RELEASES[@]}; do
+  echo "Origin: knqyf263.github.io/trivy-repo/deb
```
Comment (knqyf263) [inline ci/deploy-deb.sh:19]: We forgot to replace with aquasecurity.

```suggestion
  echo "Origin: aquasecurity.github.io/trivy-repo/deb
```

File: ci/deploy-deb.sh [hunk 2]
```diff
@@ -12,6 +12,20 @@ for release in ${DEBIAN_RELEASES[@]} ${UBUNTU_RELEASES[@]}; do
...
+Label: github.io/knqyf263
```
Comment (knqyf263) [inline ci/deploy-deb.sh:20]:
```suggestion
Label: github.io/aquasecurity
```

**Review by knqyf263 — CHANGES_REQUESTED** [review]
I want you to make sure my suggested changes don't cause any issues.

**Comment by afdesk** [comment]
> I want you to make sure my suggested changes don't cause any issues.

do you mean changes `knqyf263` to `aquasecurity`?

File: ci/deploy-deb.sh (hunk 1 above)
Comment (afdesk) [reply ci/deploy-deb.sh]: updated

File: ci/deploy-deb.sh (hunk 2 above)
Comment (afdesk) [reply ci/deploy-deb.sh]: ditto

**Comment by afdesk** [comment]
it seems these changes affect on metadata only.

**Review by knqyf263 — COMMENTED** [review]
I have another concern now. Does it drop old release supports? For example, will trusty be removed?
https://github.com/aquasecurity/trivy-repo/blob/86725c045b26843c173607ca7a6b04943fb4d1b7/deb/conf/distributions#L57-L63

**Comment by knqyf263** [comment]
I didn't know `trusty` was still supported. I meant we probably should keep old releases reaching EOL as well.

**Comment by afdesk** [comment]
> I didn't know `trusty` was still supported. I meant we probably should keep old releases reaching EOL as well.

yes, `trusty` is supported now but only with Extended Security Maintenance (ESM).

`ubuntu-distro-info` allows to select lists of `--all` versions, or `--supported`/`--unsupported` versions.

should we use `--all`?

**Comment by knqyf263** [comment]
If we remove old versions, those versions will not work suddenly, right? I'm not saying we should keep supporting, but should not remove packages for those versions.

**Comment by afdesk** [comment]
> If we remove old versions, those versions will not work suddenly, right? I'm not saying we should keep supporting, but should not remove packages for those versions.

oh, it seems i understood your idea.
then we can build configs for all distros and will include debs only for `supported` and `supported-esm`.

**Comment by afdesk** [comment]
I accidentally broke this PR, will create a new

**Comment by afdesk** [comment]
created #2404

**Pull request closed without merging**